python/pyarrow/src/arrow/python/helpers.h

```diff
@@ -47,6 +47,19 @@ ARROW_PYTHON_EXPORT PyObject* PyFloat_FromHalf(uint16_t value);
 // \brief Convert a Python object to a half-float uint16_t value.
 ARROW_PYTHON_EXPORT Result<uint16_t> PyFloat_AsHalf(PyObject* obj);
 
+/// \brief Create an array of evenly spaced values within a given interval.
+/// This function is similar to Python's `range` function.
+/// The resulting array will contain values starting from `start` up to but not
+/// including `stop`, with a step size of `step`. If `step` is zero, the function
+/// will return an empty array.
```

Comment:
A zero step should be an error instead.
```python
>>> range(1, 10, 0)
Traceback (most recent call last):
  Cell In[1], line 1
    range(1, 10, 0)
ValueError: range() arg 3 must not be zero

>>> np.arange(1, 10, 0)
Traceback (most recent call last):
  Cell In[3], line 1
    np.arange(1, 10, 0)
ZeroDivisionError: division by zero
```

Reply:
that makes sense, we return `Status::Invalid` now